hs_core/hydroshare/utils.py

```diff
@@ -121,6 +124,143 @@ def group_from_id(grp):
     return tgt
 
 
+def get_user_zone_status_info(user):
+    """
+    This function should be called to determine whether the site is in production and whether user zone
+    functionality should be enabled or not on the web site front end
+    Args:
+        user: the requesting user
+    Returns:
+        in_production, enable_user_zone where both are boolean indicating whether the site is
+        in production and whether user zone functionality should be enabled or not on the web site front end
+    """
+    if user is None:
+        return None, None
+    if not hasattr(user, 'userprofile') or user.userprofile is None:
+        return None, None
+
+
+    in_production = True if settings.IRODS_USERNAME == settings.HS_WWW_IRODS_PROXY_USER else False
+    enable_user_zone = user.userprofile.create_irods_user_account
+    if not in_production and enable_user_zone:
+        # if these settings are not empty, for example, in users' local
+        # development environment for testing, user_zone selection is shown
+        if (not settings.HS_WWW_IRODS_PROXY_USER_PWD or \
+            not settings.HS_WWW_IRODS_HOST or not settings.HS_WWW_IRODS_ZONE):
+            enable_user_zone = False
+    return in_production, enable_user_zone
+
+
+def is_federated(homepath):
+    """
+    Get size of a data object from iRODS user zone
+    Args:
+        user: the requesting user
+        fname: the logical iRODS file name with full logical path
+
+    Returns:
+    the size of the file
+    """
+    irods_storage = IrodsStorage('federated')
+    # if HS WWW iRODS proxy user can list homepath, homepath is federated; otherwise, it is not federated
+    return irods_storage.exists(homepath)
+
+
+def get_federated_zone_home_path(filepath):
+    """
+    Args:
+        filepath: the iRODS data object file path that included zone name in the format of /zone_name/home/user_name/file_path
+
+    Returns:
+        the zone name extracted from filepath
+    """
+    if filepath and filepath.startswith('/'):
+        split_path_strs = filepath.split('/')
+        # the Zone name should follow the first slash
+        zone = split_path_strs[1]
+        return '/{zone}/home/{local_proxy_user}'.format(zone=zone, local_proxy_user=settings.HS_LOCAL_PROXY_USER_IN_FED_ZONE)
+    else:
+        return ''
+
+
+def get_fed_zone_file_size(fname):
+    """
+    Get size of a data object from iRODS user zone
+    Args:
+        fname: the logical iRODS file name with full logical path
+
+    Returns:
+    the size of the file
+    """
+
+    irods_storage = IrodsStorage('federated')
+    return irods_storage.size(fname)
+
+
+def get_fed_zone_files(irods_fnames):
```

Comment:
I am OK with having a separate function to get a file from local iRODS zone over to Django. But this will be a very useful function many resource types can use ( for example, I can use it in timeseries) as we start doing some processing of the stored files more and more. Having such a function will reduce duplicate code.